components/dropdown/dropdown-content-mixin.js

```diff
@@ -477,8 +483,9 @@ export const DropdownContentMixin = superclass => class extends RtlMixin(supercl
 
 			const maxHeight = Math.floor((this.openedAbove ? spaceAround.above : spaceAround.below) - headerFooterHeight);
 			if (!this.noAutoFit && maxHeight && maxHeight > 0) {
-				this._maxHeight = maxHeight;
-				this.__toggleOverflowY(contentRect.height > maxHeight);
+				this._height = this.maxHeight !== null && maxHeight > this.maxHeight && headerFooterHeight < this.maxHeight
```

Comment:
These conditions make me a bit nervous, and to be honest it's not immediately what cases they are.  Hopefully they are captured in visual-diff tests.

Reply:
I added a comment to give some explanation and another v-diff test to cover the other case. In longer form, to sanity-check that the comment and behaviour make sense:

- We only want to apply the max height if it exists and there's enough space to need to restrict it.
- If the header and footer combined are greater than the max height, no content would show up, so we default to regular sizing to make sure content appears (the header and footer sections can't scroll so there's nothing to be done about that 🤷) Typically this shouldn't happen but it's a graceful failure in case of changes impacting header/footer content.